test/unit/component/parser/json/BpmnJsonParser.flowNode.test.ts

```diff
@@ -383,50 +381,131 @@ describe.each([
     });
   }
 
-  describe('incoming/outgoing management', () => {
-    type Input = {
-      incoming: string | string[];
-      outgoing: string | string[];
-    };
-    type Expected = {
-      incoming: string[];
-      outgoing: string[];
-    };
+  describe(`incoming/outgoing management for ${bpmnKind}`, () => {
+    const isTask = bpmnKind.toLowerCase().endsWith('task');
+    const flowNodeParameterKind = isTask ? 'task' : 'gateway';
 
     it.each`
-      title       | input                                                                                             | expected
-      ${'string'} | ${{ incoming: 'incoming_id_1', outgoing: 'outgoing_id_1' }}                                       | ${{ incoming: ['incoming_id_1'], outgoing: ['outgoing_id_1'] }}
-      ${'array'}  | ${{ incoming: ['incoming_id_1', 'incoming_id_2'], outgoing: ['outgoing_id_1', 'outgoing_id_2'] }} | ${{ incoming: ['incoming_id_1', 'incoming_id_2'], outgoing: ['outgoing_id_1', 'outgoing_id_2'] }}
-    `('should convert incoming and outgoing when defined as $title', ({ input, expected }: { input: Input; expected: Expected }) => {
-      const processJson = { ...processWithFlowNodeAsObject };
-      processJson[`${bpmnKind}`].incoming = input.incoming;
-      processJson[`${bpmnKind}`].outgoing = input.outgoing;
-      const json = {
-        definitions: {
-          targetNamespace: '',
-          process: processJson,
-          BPMNDiagram: {
-            name: 'process 0',
-            BPMNPlane: {
-              BPMNShape: {
-                id: `shape_${bpmnKind}_id_0`,
-                bpmnElement: `${bpmnKind}_id_0`,
-                Bounds: { x: 10, y: 12, width: 40, height: 45 },
-              },
+      title       | input         | expectedAttribute
+      ${'string'} | ${'incoming'} | ${'bpmnElementIncomingIds'}
+      ${'array'}  | ${'incoming'} | ${'bpmnElementIncomingIds'}
+      ${'string'} | ${'outgoing'} | ${'bpmnElementOutgoingIds'}
+      ${'array'}  | ${'outgoing'} | ${'bpmnElementOutgoingIds'}
+    `(
+      `should convert as Shape, when a process contains a ${bpmnKind} with $input attribute as $title`,
```

Comment:
**nitpick:** naming of the test. We are only testing Shape conversion in this files, so why adding it everywhere in the tests

This remark applies to all tests in this file and to other test files of this PR.